### ISSUE TITLE
Consider changing `truthy` linter's `check-keys` default

### ISSUE THREAD

**Issue by gravieure** [adrienverge/yamllint#559]
The default configuration for the `truthy` linter [applies it to keys as well as values](https://github.com/adrienverge/yamllint/blob/6bfd6756e2414e0a16a256eaf6575e922d7e2ac9/yamllint/rules/truthy.py#L43).  The syntax for [GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) definitions uses an [`on`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on) key to determine what event the workflow applies to.

The upshot is that, out of the box, yamllint will blow up on any YAML document configuring a GitHub Action.  This certainly can be fixed by disabling `check-keys`, but the presence of that option is non-obvious.

Given that GitHub Actions are a very common YAML usecase, do you think it'd be reasonable to disable this linter's `check-keys` option by default?

**Comment by adrienverge** [comment]
Hello, is it a duplicate of #540, #430, #344, #232, #247, #158 and others?

**Comment by gravieure** [comment]
No; those all assume that the behavior is a bug needing to be fixed.  I understand that it is not a bug, but an option which is enabled by default.  My question is: would it make sense to change that default?  The numerous issues filed on this subject are strong evidence that the default is confusing.

If changing the default is disagreeable, what do you think about the `truthy` linter printing some info about the `check-keys` option in this scenario?  I think that would also reduce confusion and duplicate bug reports, since people would be guided to the right option without needing to file an issue about it.

**Comment by SubaruArai** [comment]
I was thinking to suggest using a `%YAML 1.2` tag to automatically disable `check-keys`.
But then again, found out that `pyyaml` (still!) doesn't support yaml 1.2.

Switching to `ruamel.yaml` might be an option, but that's for another discussion.

**Comment by 777arc** [comment]
I think this is very reasonable, either disable check-keys or disable it for files inside of .github/workflows

**Comment by adrienverge** [comment]
TL;DR: For those seeking a quick-win to silent the error on GitHub Actions, here are 4 solutions:
\- use [config option](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.truthy) `rules: {truthy: {check-keys: false}}`
\- use [config option](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.truthy) `rules: {truthy: {allowed-keys: [on, …]}}`
\- disable the check on the specific line with `# yamllint disable-line rule:truthy`
\- quote the `on` key, e.g. `"on"`

> No; those all assume that the behavior is a bug needing to be fixed. I understand that it is not a bug, but an option which is enabled by default. My question is: would it make sense to change that default? The numerous issues filed on this subject are strong evidence that the default is confusing.

I understand, but changing a default configuration in yamllint would break behavior in processes (e.g. continuous integration) for many users and companies. It is not an option to consider.

> If changing the default is disagreeable, what do you think about the `truthy` linter printing some info about the `check-keys` option in this scenario? I think that would also reduce confusion and duplicate bug reports, since people would be guided to the right option without needing to file an issue about it.

This is interesting. But if I understand correctly, your proposition would produce extra "human-friendly" output amongst yamllint regular output (which has a constant pattern). I fear that this also would break tools relying on yamllint's output.

**Comment by sandstrom** [comment]
Another option:


```yml
  # don't use YAML's yes/no as boolean values (common cause of bugs)
  truthy:
    level: error

    # github action workflows use the key 'on' without quotes
    ignore: |
      .github/workflows/*.yml
```

**Comment by Nothing4You** [comment]
I've seen it suggested in a few issues in this repo to prefix GHA workflow files with `%YAML 1.2`, but this doesn't work with GHA, as they do not recognize this syntax. The workflow just errors.

**Comment by sandstrom** [comment]
@Nothing4You Do you know if the `---` has been used too?

I think the syntax is:

```
%YAML 1.2
---

# rest of yaml workflow here…

```

If Github doesn't support it, it would be a good idea to send that as a feature suggestion to Github. Preferably from paid accounts, because I don't think they answer questions from free Github accounts (which is reasonable).

**Comment by Nothing4You** [comment]
Yes, it has been used as well.
See https://github.com/Nothing4You/yamllint-issue-559/tree/main/.github/workflows and https://github.com/Nothing4You/yamllint-issue-559/actions

**Comment by sandstrom** [comment]
@Nothing4You Thanks! I've written to Githubs support reporting this as a bug.

**Comment by webknjaz** [comment]
> - use [config option](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.truthy) `rules: {truthy: {allowed-keys: [on, …]}}`

This made me think that `allowed-keys` is implemented in yamllint. But it's not. Only `allowed-values` exists... It'd be nice to actually have it so that it could be applied to YAML keys but not values, for example.
